build/dsc/templates/dsc_type.rb.erb

```diff
@@ -215,6 +225,9 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
       value.kind_of?(Hash) ?
         [PuppetX::Dsc::TypeHelpers.munge_embeddedinstance(mof_type_map, value)] :
         value.map { |v| PuppetX::Dsc::TypeHelpers.munge_embeddedinstance(mof_type_map, v) }
+<%     elsif (property.int_array? || property.uint_array?) -%>
```

Comment:
This can never happen.  When dealing with embedded instances, the corresponding MOF type is always either `string` or `string[]`